### ISSUE TITLE
styled-jsx opt-out?

### ISSUE THREAD

**Issue by Cinamonas** [zeit/next.js#2384]
Is there currently a way to completely disable `styled-jsx` that's bundled with Next.js? If not, it would be great to have this option as currently it sometimes clashes with other CSS solutions by doing preprocessing of its own.

See #2304 and I've also been having issues with importing plain CSS through [`wrap-in-js`](https://github.com/davibe/babel-plugin-wrap-in-js).

**Comment by arunoda** [comment]
There's no need to opt-out. It's already opt-in.
If you don't use it, it won't included in the bundle.

**Comment by Cinamonas** [comment]
@arunoda, maybe I'm missing something, but I'm not using it anywhere explicitly, and it still parses (and breaks) the stylesheet that I'm importing directly from `node_modules` (which I then `injectGlobal` through styled-components).

Right now, this is how I opt-out:

```js
// babel-preset.js
const preset = require('next/babel');

preset.plugins = removeStyledJsx(preset.plugins);

module.exports = preset;

function removeStyledJsx(plugins) {
  return plugins.filter(plugin => plugin.indexOf('styled-jsx') === -1);
}

// .babelrc
{
  "presets": [
    "./babel-preset"
  ],
  // …
}
```

And this fixes the issues that I'm having.

**Comment by timneutkens** [comment]
@giuseppeg ^

**Comment by giuseppeg** [comment]
@timneutkens @arunoda there is nothing I can do to prevent that from happening.
See https://github.com/styled-components/styled-components/issues/934

Seems like that they released a new version of `babel-plugin-styled-components` with the workaround that I suggested.

**Comment by arunoda** [comment]
@Cinamonas okay even if it's not included in the bundle, it parses. Good point.
Thanks @giuseppeg  for the pointer.

I hope with the new version, we can get rid of this issue.

**Comment by timneutkens** [comment]
@giuseppeg I was expecting you had seen this issue before 😄 Thanks for clarifying ❤️

**Comment by Sly777** [comment]
On the last versions of next.js-styled comp-babel styled comp, There is no problem. I tried on my package (https://github.com/Sly777/ran) and It works without getting error. FYI, @arunoda :)

**Comment by timneutkens** [comment]
Awesome @Sly777 so we can close this? Or does anyone want to confirm that it's fixed 👌 We might have to update the example 🤔

**Comment by Cinamonas** [comment]
@timneutkens, I've just upgraded to the latest beta, removed my `styled-jsx` out-out code and everything seems to be working 🎉

**Comment by arunoda** [comment]
Awesome. Let's close this then.

**Comment by dortzur** [comment]
@Cinamonas this also solves `Nesting detected at 4937:239. Unfortunately nesting is not supported by styled-jsx` when using [with-global-stylesheet](https://github.com/zeit/next.js/tree/v3-beta/examples/with-global-stylesheet)
Thanks!

**Comment by giuseppeg** [comment]
fwiw I fixed this in styled-jsx `1.0.9`
